src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -367,6 +365,13 @@ export class MeetingManager implements AudioVideoObserver {
       return;
     }
 
+    // When an attendee leaves, we remove AudioVideoObservers and nullify AudioVideoFacade and MeetingSession object.
+    // This results into missing few meeting events triggered with audioVideoDidStop such as meetingEnded, meetingFailed and meetingStartFailed.
+    // We may also loose audioInputUnselected and videoInputUnselected events as we choose null devices when an attendee leaves.
+    // When a new AudioVideoFacade object is created remove and re-add the eventDidReceive observer which wont leak.
+    this.eventDidReceiveObserver && this.audioVideo.removeObserver(this.eventDidReceiveObserver);
```

Comment:
In case if you didn't see it, we could remove this condition check here.

Reply:
Yup I have removed it, thanks!